### PR TITLE
aro-3168: retry on conflict and not found for disabling samples

### PR DESCRIPTION
**Pull request by s-amann**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-3168

### What this PR does / why we need it:

This step can fail if the RP gains control over the creating cluster before this CR is applied

### Test plan for issue:

Unit tests

### Is there any documentation that needs to be updated for this PR?

no

### CODE REVIEW

File: pkg/cluster/samples.go [hunk 1]
```diff
@@ -19,17 +20,20 @@ func (m *manager) disableSamples(ctx context.Context) error {
...
 
-	return retry.RetryOnConflict(retry.DefaultRetry, func() error {
-		c, err := m.samplescli.SamplesV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
-		if err != nil {
-			return err
-		}
+	return retry.OnError(retry.DefaultRetry,
+		func(err error) bool {
+			return errors.IsConflict(err) || errors.IsNotFound(err)
+		}, func() error {
```
Comment (SudoBrendan) [inline pkg/cluster/samples.go:26]: nit: I was confused by this at first glance, can we fix the formatting to have one parameter per-line considering we're doing in-line anonymous functions (that is, huge, formatted parameters)?

```
return retry.OnError(
    a,
    b,
    c
)
```

**Review by SudoBrendan — APPROVED** [review]
Overall LGTM, one nit.

File: pkg/cluster/samples.go (hunk 1 above)
Comment (s-amann) [reply pkg/cluster/samples.go:26]: totally, I'll update with re-formatting.

File: pkg/cluster/samples.go [hunk 2]
```diff
@@ -19,17 +20,22 @@ func (m *manager) disableSamples(ctx context.Context) error {
...
+			return errors.IsConflict(err) || errors.IsNotFound(err)
+		},
+		func() error {
+			c, err := m.samplescli.SamplesV1().Configs().Get(ctx, "cluster", metav1.GetOptions{})
+			if err != nil {
+				return err
+			}
 
-		c.Spec.ManagementState = operatorv1.Removed
+			c.Spec.ManagementState = operatorv1.Removed
```
Comment (SrinivasAtmakuri) [inline pkg/cluster/samples.go:34]: LGTM, a minor nit, GitHub is highlighting whitespace here, maybe this should be a tab space?

**Review by petrkotas — APPROVED** [review]
@s-amann Thanks! LGTM

**Pull request merged**